server/src/test/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDeciderUnitTests.java

```diff
@@ -901,6 +904,74 @@ public void testDecidesYesIfWatermarksIgnored() {
         assertThat(decision.getExplanation(), containsString("disk watermarks are ignored on this index"));
     }
 
+    public void testDecidesYesIfPromotableShardRole() {
+        final Metadata metadata = Metadata.builder()
+            .put(IndexMetadata.builder("test").settings(settings(IndexVersion.V_8_10_0)).numberOfShards(1).numberOfReplicas(0))
```

Comment:
Is IndexVersion important and would it be kept up to date in the future?

Reply:
Thanks for catching this, changed to `IndexVersion.current()`